### ISSUE TITLE
Document configs

### ISSUE THREAD

**Issue by zuiderkwast** [valkey-io/valkey-doc#154]
Currently, we just refer to valkey.conf (the template config file) where the fields are described in comments. This isn't very good documentation.

Document each config parameter in the proper page [valkey.conf.md](https://github.com/valkey-io/valkey-doc/tree/main/topics/valkey.conf.md). Instead of referring to the config files of different versions, we can mention for each field in which version it was introduced.

**Comment by rjd15372** [comment]
I also think this is a good idea. Having the config options description in our documentation is the way to go. But there's a maintenance burden here that we need to decide how to tackle on, because we will have duplicate information (in the valkey.conf file, and here in the documentation).

My take on this is that whenever a developer opens a PR with changes to the `valkey.conf` file in the valkey repo, the reviewers should ask the developer to also open a PR in the valkey-doc repo with updating the documentation. This way we distribute the burden of maintenance between all developers.

**Comment by zuiderkwast** [comment]
Yes, for any feature that requires doc changes, we ask the developer to do it and we add the tag 'doc-pr-needed' to the PR.

Can I assign you to this issue? ;)

**Comment by rjd15372** [comment]
@zuiderkwast you can assign it to @nastena1606

**Comment by zuiderkwast** [comment]
Ideas:

1. Minimal. Change the link to `valkey.conf` to point to valkey.conf in unstable branch. Remove the old links to Valkey 7 config files and old redis.conf.
2. Automated. We could write a script to include the contents of `valkey.conf` (the example config with comments) in a code block on this page. In this way, contributors that modify configs don't need to make a doc PR. We only need to run the script when we make a new release.
3. Full. Add proper documentation of the configs in this file. Some drawbacks: We will have duplicated information (new docs + the comments valkey.conf) and contributors will have to make a doc PR when changing some configs.

**Comment by melroy89** [comment]
> What we talked about in weekly core meeting:
> 
>     1. We want to pull out the documentation from the valkey.conf file. Where and how is not exactly defined yet, but we would like to have it stored in markdown/json/yaml/toml somewhere else.
> 
>     2. In Valkey 10 we will stop updating the valkey.conf file. We will find a way to keep them in sync with each other, the valkey.conf file will not be authoritative.

sounds like a good plan.
